apis/nucleus/src/components/listbox/ListBox.jsx

```diff
@@ -231,9 +239,11 @@ export default function ListBox({
     // If values have been filtered in the currently loaded page, we want to
     // prevent rendering empty rows by assigning the actual number of items to render
     // since count (qcy) does not reflect this in DQ mode currently.
-    const hasFilteredValues = ps.some((page) => page.qArea.qHeight < MINIMUM_BATCH_SIZE);
+    // If any qMatrix was not 100 length && result of it was less that qcy
+    // then => qTop + qHeight might indicate length of items as long as you scroll and fetch more items
+    const notEnoughItems = ps.some((p) => p.qArea.qHeight !== MINIMUM_BATCH_SIZE);
     const h = Math.max(...ps.map((page) => page.qArea.qTop + page.qArea.qHeight));
-    return hasFilteredValues ? h : count;
+    return h < count && notEnoughItems ? h : count;
```

Comment:
Count is not reliable in DQ mode, why use it here?

Reply:
since it is the original amount of items in list, we are checking if current height is less that than, then we should detect that it is end of the list and should not expect any more items

Reply:
I understand what you are trying to do, but if count (or qcy) is not reliable in DQ mode then we should not use it. What if it is lower than the actual amount of items?

Reply:
it won't because Vlad has mentioned that it will always represent the original amount of list items 
we cant filter a list and have more results than the original list items

Reply:
Then I don't see how this changes any logic. This would only be the case when there are no items filtered at all, in which case it would also return the count

Reply:
No actually, because we will return the value of `h` only in last pages of api calls and based on that we will return the correct list items.